### ISSUE TITLE
AttributeError: Method VideoInput is not a registered operator. Did you mean: []

### ISSUE THREAD

**Issue by 953585130** [facebookresearch/R2Plus1D#44]
Hi,
Thank you for the open source code provided! I have installed gpu support as per the tutorial, and I get an error when I execute the script that extracts features.

AttributeError: Method VideoInput is not a registered operator. Did you mean: []

I tried to debug or reported the same error, I hope to get your help!

**Comment by kateiyas** [comment]
I get the same error after successfully installing caffe2 and pytorch in a local anaconda installation with: 

`conda install pytorch-nightly torchvision -c pytorch`

Would be great to get help here.

**Comment by 953585130** [comment]
@kateiyas 
Thank you for your reply, but I didn't understand what you mean. Do you mean that you used anaconda to install caffe2 and pytorch and the same error occurred? But what should be done? I searched the Internet for a long time, probably said that it may be related to opencv, I tried to recompile the version but still have the same problem. I really don't know how to solve this problem.

**Comment by kateiyas** [comment]
@953585130: Yes, I got the same error `AttributeError: Method VideoInput is not a registered operator. Did you mean: []`. The error was thrown after following the tutorial for feature extraction, calling `extract_features.py` (which I mention in my reply). I have also described it [here](https://github.com/pytorch/pytorch/issues/12401).

**Comment by 953585130** [comment]
@kateiyas 
I have solved this problem and I have re-downloaded other versions of pytorch, which supports VideoInput. You can try another version, the version given by the tutorial has a problem or it was not fully cloned when cloning.

**Comment by greathope** [comment]
@953585130  can you please tell me which version of pytorch you are using?

**Comment by orionr** [comment]
The default binary builds are generated without OpenCV and ffmpeg dependency, which means that these operators (all of those under caffe2/video/) aren't built. You'll need to build from source to have these operators available.

We should update the R2Plus1 tutorial to make a note of this.

**Comment by 953585130** [comment]
@greathope @kateiyas 
Starting with a new system, I installed cuda8.0 and cudnn6.0 directly after installing the graphics driver. The next steps I followed were tutorials installed step by step.

Follow the tutorial to clone pytorch, the entire pytorch is about 300MB. I searched online for a tutorial on installing caffe2, one of which has a pytorch zip file, which is 400MB after decompression. Then I compiled it with this and did not report an error when performing feature extraction.

I don't know if your email supports file transfer? If your mailbox supports file transfer function, please send me your full email address, I can send this compressed package to you.

Just orionr replied to this question, you can look at his explanation.

**Comment by greathope** [comment]
@953585130 could you please send that file to my email? my email is zhangmiaoatzju@gmail.com or zhangmiaoatzju@163.com, many thanks!!!

**Comment by 953585130** [comment]
@greathope 
I have sent it to your 163 mailbox, you can try to solve your problem.

**Comment by 953585130** [comment]
@greathope 
Do you have qq, we can discuss it together.

**Comment by greathope** [comment]
@953585130 sure, my qq is 1215714511~

**Comment by kateiyas** [comment]
Just a note for people stumbling over this thread: As of to date, building pytorch from source with `USE_FFMPEG=1` and `USE_OPENCV=1` does **not** build the Caffe2 video operators that are necessary for R2Plus1D to work. Theoretically building pytorch (+caffe2) from source via `USE_FFMPEG=1 USE_OPENCV=1 USE_LEVELDB=0 USE_LMDB=1 python setup.py install` should solve the issue in the thread, but it doesn't. Could make sense to reopen this issue. 

@953585130 Do you still have a link to the online tutorial you mention? Otherwise: muellerkatj gmail com – I would be very grateful for help in this matter.

**Comment by 953585130** [comment]
@kateiyas 
The compressed file has been delivered to your mailbox, and the rest will be installed according to the tutorial.

**Comment by orionr** [comment]
Also, @kateiyas, a heads up that `USE_FFMPEG` wasn't passed through `setup.py` until https://github.com/pytorch/pytorch/pull/12543 landed, which was yesterday. If you could retest with latest master (`git pull; git submodule update`) that would be great, but feel free to use the compressed file as well.

**Comment by dutran** [comment]
@kateiyas can you try the latest pytorch master as mentioned by @orionr , if it does not work, we can re-open it. thanks.

**Comment by kateiyas** [comment]
@dutran @orionr The new master seemed to get me around the video operator bug.

(Running feature extraction still did not work because R2Plus1D will only work with GPU support, i.e. my `NO_GPU=1` is not possible. If I compile with GPU support I get a segmentation fault now. )

**Comment by 953585130** [comment]
@shanshuo 
Has been sent to your mailbox.

**Comment by lrd8715923** [comment]
@953585130 
Hi could you also send me the compressed pytorch file.I can't build it sucess.
my emil:424079971@qq.com
Thanks.

**Comment by ChisamXz** [comment]
@953585130 
Hi, could you also send me the pytorch file? I've really suffered enough from the installation.
I'll really appreciate the sharing!

My email is: **chisam.xiz@gmail.com**

Thanks

**Comment by 953585130** [comment]
@ChisamXz 
I will send it to your email later.

**Comment by yimikai** [comment]
> @kateiyas
> I have solved this problem and I have re-downloaded other versions of pytorch, which supports VideoInput. You can try another version, the version given by the tutorial has a problem or it was not fully cloned when cloning.

@953585130  hi, bro,  could you please send that file to my email? my email is 1544788527@qq.com or hudongfang06@163.com, many thanks!!!

**Comment by vinitmuchhala** [comment]
Hey, can you send me an email as well? it is vinit.muchhala@hotmail.com thanks

**Comment by oLIVIa-Ld** [comment]
@953585130 hi, could you please send me an email for the correct pytorch? thank you, so much. oliviald@126.com

**Comment by Jinyong-Huang** [comment]
@953585130 I tried to install pytorch/caffe2 for a few days, but failed.could you please send that file to my email? my email is 543401923@qq.com

**Comment by R2D2oid** [comment]
I followed the tutorial instructions and after resolving a whole lot of other issues along the way still, get the infamous **VideoInput** error!

@953585130 @greathope @kateiyas or others,  do you mind sharing a link to the built package? maybe on google drive or elsewhere?

**Comment by Weed-eat-meat** [comment]
@953585130   Hi，I had the same problrm, can you send me an email as well? it is syy960315@163.com. thanks

**Comment by Yuqing1119** [comment]
@953585130 Hi, I had the same problem, can you send me an email for the correct pytorch as well?  It is syuqing@ruc.edu.cn, thank you very much!

**Comment by NemioLipi** [comment]
@953585130 Hi, I am facing the same problem, would you please send me the correct pytorch version as well? My email address is nemiolipi@gmail.com,

**Comment by Yuqing1119** [comment]
@NemioLipi Hi, I have solved the problem by getting back to pytorch version v1.1.0. The script is as follows:

git clone https://github.com/pytorch/pytorch.git
cd pytorch && git checkout v1.1.0 && git submodule update --init --recursive

But to be noted, the 487 line of caffe2/video/video_input_op.h should be replaced with the following: 
ReinitializeTensor(&prefetched_label_, vector<int64_t>(1, batch_size_ * clip_per_video_ * multi_crop_count_), at::dtype<int>().device(CPU));
Otherwise it will throw errors when compiling.

Then, USE_OPENCV=1 USE_FFMPEG=1 USE_LMDB=1 USE_CUDA=1 python setup.py install

**Comment by nassimaNoufail** [comment]
@953585130 Hi, I have the same issue can you please email me the correct PyTorch version, my email: nassima.noufail@gmail.com Thank you!!

**Comment by Tracy-Lee1993** [comment]
@953585130 Hi, I have the same issue can you please email me the correct PyTorch version，my email：lizhaoyang_1993@qq.com
